### ISSUE TITLE
[Bug]: IllegalResponseException when query node is restarting in cluster and search is performed

### ISSUE THREAD

**Issue by karanbhatia-jm** [milvus-io/milvus#18544]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.0.2
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): Java SDK 2.0.4
- OS(Ubuntu or CentOS): NA
- CPU/Memory: NA
- GPU: NA
- Others: NA
```

### Current Behavior

When query node is restarting on Kubernetes cluster (deployed using helm) and a similarity search is performed it throws `IllegalResponseException: Result ids is illegal`. Even after the pod is in running state the same exception is thrown. I checked the query node logs. It seems that the segmentIds are not getting populated correctly and hence the empty result response. 

### Expected Behavior

When the pod is back to the running state correct search results should be returned.

### Steps To Reproduce

```markdown
1. Deploy Milvus cluster using the official helm chart on Kubernetes
2. Feed some dummy vector data to the database (~10,000 512 dimensional vectors)
3. Once all the pods are in running state, kill the query node pod
4. Make a similarity search request using any dummy vector (512 dimensional) while the query node pod is restarting. (Index params: {metric_type: IP, indexType: FLAT})
```

### Milvus Log

```
[2022/08/05 08:33:50.587 +00:00] [DEBUG] [impl.go:37] ["Get QueryNode component states"]
[2022/08/05 08:33:50.587 +00:00] [DEBUG] [impl.go:62] ["Get QueryNode component state done"] [stateCode=Healthy]
[2022/08/05 08:33:52.270 +00:00] [DEBUG] [impl.go:37] ["Get QueryNode component states"]
[2022/08/05 08:33:52.270 +00:00] [DEBUG] [impl.go:62] ["Get QueryNode component state done"] [stateCode=Healthy]
[2022/08/05 08:34:01.593 +00:00] [DEBUG] [query_collection.go:350] ["check if query timeout"] [collectionID=435073729928101889] [msgID=435075706389352196] [TimeoutTs=0] [curTime=435077366832955392] [timeoutTsPhysical=1970/01/01 00:00:00.000 +00:00] [curTimePhysical=2022/08/05 08:34:01.593 +00:00]
[2022/08/05 08:34:01.593 +00:00] [DEBUG] [time_recorder.go:78] ["receiveQueryMsg 435075706389352196: get searchable time done (0ms)"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [query_collection.go:557] ["doing query in receiveQueryMsg..."] [collectionID=435073729928101889] [sm.GuaranteeTimestamp=1970/01/01 00:00:00.000 +00:00] [serviceTime=2022/08/05 08:34:01.303 +00:00] ["delta seconds"=18011666706] [msgID=435075706389352196] [msgType=search]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [query_collection.go:1082] ["historical search start"] [msgID=435075706389352196]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [historical.go:124] ["no partition specified, search all partitions"] [collectionID=435073729928101889] ["all partitions"="[435073729928101890]"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [query_collection.go:1088] ["historical search"] [msgID=435075706389352196] [collectionID=435073729928101889] ["searched partitionIDs"="[435073729928101890]"] ["searched segmentIDs"="[]"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [time_recorder.go:78] ["search 435073729928101889(nq=1, k=30), msgID = 435075706389352196: historical search done, msgID = 435075706389352196 (0ms)"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [query_collection.go:1091] ["streaming search start"] [msgID=435075706389352196]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [streaming.go:125] ["no partition specified, search all partitions"] [collectionID=435073729928101889] [vChannel=by-dev-rootcoord-dml_1_435073729928101889v1] ["all partitions"="[435073729928101890]"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [streaming.go:163] ["get segmentIDs by vChannel"] [collectionID=435073729928101889] [vChannel=by-dev-rootcoord-dml_1_435073729928101889v1] [partitionID=435073729928101890] [segmentIDs="[]"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [query_collection.go:1098] ["streaming search"] [msgID=435075706389352196] [collectionID=435073729928101889] ["searched dmChannel"=by-dev-rootcoord-dml_1_435073729928101889v1] ["searched partitionIDs"="[435073729928101890]"] ["searched segmentIDs"="[]"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [streaming.go:125] ["no partition specified, search all partitions"] [collectionID=435073729928101889] [vChannel=by-dev-rootcoord-dml_0_435073729928101889v0] ["all partitions"="[435073729928101890]"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [streaming.go:163] ["get segmentIDs by vChannel"] [collectionID=435073729928101889] [vChannel=by-dev-rootcoord-dml_0_435073729928101889v0] [partitionID=435073729928101890] [segmentIDs="[]"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [query_collection.go:1098] ["streaming search"] [msgID=435075706389352196] [collectionID=435073729928101889] ["searched dmChannel"=by-dev-rootcoord-dml_0_435073729928101889v0] ["searched partitionIDs"="[435073729928101890]"] ["searched segmentIDs"="[]"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [time_recorder.go:78] ["search 435073729928101889(nq=1, k=30), msgID = 435075706389352196: streaming search done, msgID = 435075706389352196 (0ms)"]
[2022/08/05 08:34:01.594 +00:00] [DEBUG] [query_collection.go:1128] ["QueryNode Empty SearchResultMsg"] [collectionID=435073729928101889] [msgID=435075706389352196] [vChannels="[by-dev-rootcoord-dml_1_435073729928101889v1,by-dev-rootcoord-dml_0_435073729928101889v0]"] [sealedSegmentSearched="[]"]
[2022/08/05 08:34:01.595 +00:00] [DEBUG] [proxy_session_manager.go:108] ["success to send search result"] [node=161] [base="msg_type:SearchResult msgID:435075706389352196 timestamp:435077366822469633 sourceID:161 "]
[2022/08/05 08:34:01.595 +00:00] [DEBUG] [time_recorder.go:78] ["search 435073729928101889(nq=1, k=30), msgID = 435075706389352196: publish empty search result done, msgID = 435075706389352196 (1ms)"]
[2022/08/05 08:34:01.595 +00:00] [DEBUG] [time_recorder.go:78] ["search 435073729928101889(nq=1, k=30), msgID = 435075706389352196: all done, msgID = 435075706389352196 (1ms)"]
[2022/08/05 08:34:01.595 +00:00] [DEBUG] [time_recorder.go:78] ["receiveQueryMsg 435075706389352196: operation done (1ms)"]
[2022/08/05 08:34:01.595 +00:00] [DEBUG] [query_collection.go:589] ["do query done in receiveQueryMsg"] [collectionID=435073729928101889] [msgID=435075706389352196] [msgType=search]
```

### Anything else?

This behaviour goes away once the data is added again to the database.

**Comment by xiaofan-luan** [comment]
hi @karanbhatia-jm 
How much data did you ingest? There seems to have no sealed segment , which means you have very little data in the system, does that make sense to you?

**Comment by karanbhatia-jm** [comment]
Data ingested is about 10500 512-dimensional float vectors

**Comment by karanbhatia-jm** [comment]
@xiaofan-luan Less data would not affect the search right?

**Comment by yanliang567** [comment]
@karanbhatia-jm any chance to reproduce it on v2.1.0? if yes, pplease refer this [script](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the whole Milvus logs for investigation

/assign @karanbhatia-jm 
/unassign

**Comment by karanbhatia-jm** [comment]
> @karanbhatia-jm any chance to reproduce it on v2.1.0? if yes, pplease refer this [script](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the whole Milvus logs for investigation
> 
> /assign @karanbhatia-jm /unassign

@yanliang567 This got solved in v2.1.0. Cannot reproduce the same bug

**Comment by yanliang567** [comment]
great to hear that, thank you for updates
